File: Bee Pilot/Bee.rb
#
#  Bee.rb
#  Bee Pilot
#
#  Created by Thomas R. Koll on 24.04.11.
#  Copyright 2011 Ananasblau.com. All rights reserved.
#


class Bee
  attr_accessor :position
  attr_accessor :path
  attr_accessor :returning

  def initialize(point)
    self.position = point || NSPoint.new(0,0)
    self.returning = false
  end
end

File: Bee Pilot/game/Bee.js
/*
  Created by Thomas R. Koll on 24.04.11.
  Copyright 2011 Ananasblau.com. All rights reserved.
*/var Bee;
Bee = (function() {
  function Bee(position) {
    this.position = position;
    this.path;
    this.speed = 1;
    this.cargo = [];
  }
  Bee.prototype.tick = function() {
    if (path[0] && self.position === path[0]) {
      return path.shift;
    }
  };
  Bee.prototype.draw = function() {};
  return Bee;
})();